packages/caliper-core/test/manager/caliper-engine.js

```diff
@@ -0,0 +1,139 @@
+/*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+const sinon = require('sinon');
+const chai = require('chai');
+const expect = chai.expect;
+
+describe('CaliperEngine', function() {
+
+    describe('Initialization', function() {
+        it('should initialize with given configurations and adapter factory', function() {
+            // TODO: Implement test
+        });
+
+        it('should set the workspace and initial return code', function() {
+            // TODO: Implement test
+        });
+    });
+
+    describe('Run Method Execution Flow', function() {
```

Comment:
Still don't like putting implementation details in the test so please change this. This section is about when a benchmark is run